src/settings.ts

```diff
@@ -0,0 +1,104 @@
+/* --------------------------------------------------------------------------------------------
+ * SonarLint for VisualStudio Code
+ * Copyright (C) 2017-2022 SonarSource SA
+ * sonarlint@sonarsource.com
+ * Licensed under the LGPLv3 License. See LICENSE.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import * as VSCode from 'vscode';
+import { ExtensionContext } from 'vscode';
+
+async function hasUnmigratedConnections(sqConnections: SqConnection[],
+                                        secureStorage: SonarLintSecretStorage): Promise<boolean> {
+  for (const connection of sqConnections) {
+    if (!await secureStorage.hasTokenForServer(connection.serverUrl)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+export function getSonarLintConfiguration(secureStorage?: SonarLintSecretStorage): VSCode.WorkspaceConfiguration {
+  const settingsFromConfigFiles = getSettingsFileConfig();
+  (async () => migrateSettings(settingsFromConfigFiles, secureStorage))();
+  return settingsFromConfigFiles;
+}
+
+async function migrateSettings(settings: VSCode.WorkspaceConfiguration, secureStorage?: SonarLintSecretStorage) {
```

Comment:
I think this function should be called `migrateConnectedModeSettings` or similar,  to describe immediately that only connections are being migrated. But it's a minor detail